kernel/filesystem/TempFs.cpp

```diff
@@ -0,0 +1,345 @@
+    { return true; (void)node; (void)context; } //no-op
+
+    size_t TempFs::ReadWrite(Node* node, const RwPacket& packet, const FsContext& context)
+    {
+        VALIDATE(node != nullptr, 0, "Node is null");
+        VALIDATE(node->type == NodeType::File, 0, "Node is not a file");
+        VALIDATE(node->driverData != nullptr, 0, "Node fsdata is null");
+        VALIDATE(packet.buffer != nullptr, 0, "Buffer is null");
+        VALIDATE(packet.length > 0, 0, "Op length of zero.");
+        (void)context; //TODO: 
+
+        size_t misalignment = packet.offset % FileCacheUnitSize();
+        size_t opSize = 0;
+        if (packet.write) //TODO: this function can probably be reduced
+        {
+            node->lock.WriterLock();
+            TempFsData* data = static_cast<TempFsData*>(node->driverData);
+
+            //handle a change in the file size (expansion or truncation if requested)
+            data->props.size = sl::Max(packet.offset + packet.length, data->props.size);
+            if (packet.truncate && packet.offset + packet.length < data->props.size)
+            {
+                data->props.size = packet.offset + packet.length;
+                //TODO: drop caches beyond new end of file
+            }
+
+            const size_t opSizeLimit = sl::Min(packet.offset + packet.length, data->props.size);
+            while (opSize < opSizeLimit)
+            {
+                const size_t copyLength = sl::Min(FileCacheUnitSize(), opSizeLimit - opSize) - misalignment;
+                auto cache = GetFileCache(node->link.cache, packet.offset + opSize, true);
+                ASSERT(cache.Valid(), "FileCache returned invalid handle.");
+                sl::memcopy(packet.buffer, opSize, AddHhdm(cache->physBase), misalignment, copyLength);
+
+                opSize += copyLength;
+                misalignment = 0; //only the first entry can be misaligned
+            }
+
+            node->lock.WriterUnlock();
+        }
+        else
+        {
+            node->lock.ReaderLock();
+            
+            const TempFsData* data = static_cast<TempFsData*>(node->driverData);
+            const size_t opSizeLimit = sl::Min(packet.offset + packet.length, data->props.size);
+
+            while (opSize < opSizeLimit)
+            {
+                const size_t copyLength = sl::Min(FileCacheUnitSize(), opSizeLimit - opSize) - misalignment;
+                auto cache = GetFileCache(node->link.cache, packet.offset + opSize, false);
+                ASSERT(cache.Valid(), "FileCache returned invalid handle.");
+                sl::memcopy(AddHhdm(cache->physBase), misalignment, packet.buffer, opSize, copyLength);
+
+                opSize += copyLength;
+                misalignment = 0; //only the first entry can be misaligned
+            }
+
+            node->lock.ReaderUnlock();
+        }
+        
+        return opSize;
+    }
+
+    bool TempFs::Flush(Node* node)
+    { return false; (void)node; } //no-op
```

Comment:
does it make sense to cast  node after returnin? (i see that probaly is a temp function) 
Or is to suppress some warning?